### PR TITLE
Fix the IT module depends on the tests module

### PR DESCRIPTION
**Pull request by res-life**
Fix the IT module depends on the tests module
This contributes #3934 

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
Remove the dependency
Move the test cases in the IT module to the tests module
Fix a random hang issue.

File: integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/mortgage/MortgageSparkSuite.scala [hunk 1]
```diff
@@ -63,6 +63,10 @@ class MortgageSparkSuite extends FunSuite {
     builder.getOrCreate()
   }
 
+  override def afterAll() = {
```
Comment (firestarman) [inline integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/mortgage/MortgageSparkSuite.scala:66]: Could you add a comment to explain what happened and why did this change ?

File: integration_tests/pom.xml [hunk 2]
```diff
@@ -64,28 +64,14 @@
...
         </dependency>
-        <!-- use aggregator jar because accessing internal classes -->
-        <dependency>
-            <groupId>com.nvidia</groupId>
-            <artifactId>rapids-4-spark-aggregator_${scala.binary.version}</artifactId>
-            <version>${project.version}</version>
-            <classifier>${spark.version.classifier}</classifier>
-            <scope>test</scope>
-        </dependency>
+
```
Comment (firestarman) [inline integration_tests/pom.xml:67]: NIT: This empty line is useless.

**Comment by res-life** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**